src/overlays/actors/ovl_Eff_Dust/z_eff_dust.c

```diff
@@ -22,28 +29,392 @@ const ActorInit Eff_Dust_InitVars = {
     (ActorFunc)EffDust_Update,
     (ActorFunc)EffDust_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Eff_Dust/func_8099D8D0.s")
+static Gfx* D_8099EB60[] = { 0xDF000000, 0x00000000, 0x00000000, 0x00000000 };
+
+enum DustEffect { Effect_0 = 0, Effect_1 = 1, Effect_2 = 2, Effect_3 = 3, Effect_4 = 4 };
+
+void EffDust_setUpdateFunc(EffDust* this, EffDustActionFunc callback_updateFunc) {
+    this->updateFunc = callback_updateFunc;
+}
+
+void EffDust_setDrawFunc(EffDust* this, EffDustActionFunc callback_drawFunc) {
+    this->drawFunc = callback_drawFunc;
+}
+
+// Members initializer (?)
+void EffDust_InitPosAndDistance(EffDust* this) {
+    s32 i;
+
+    for (i = 0; i < 0x40; i++) {
+        this->initialPositions[i].z = 0.0f;
+        this->initialPositions[i].y = 0.0f;
+        this->initialPositions[i].x = 0.0f;
+
+        this->distanceTraveled[i] = 1.0f;
+    }
+
+    this->index = 0;
+}
+
+void EffDust_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EffDust* this = THIS;
+    enum DustEffect dust_effect;
```

Comment:
```suggestion
```
We dont need to have this here since its already in the header

Reply:
What do you mean by that? This is a variable.